.pre_commit/generate-facets/generate.py

```diff
@@ -0,0 +1,253 @@
+            class_name="ClassToBeSkipped",
+            use_field_description=True,
+            use_double_quotes=True,
+            keep_model_order=True,
+            custom_template_dir=templates_location,
+            extra_template_data=defaultdict(dict, merge_dicts(extra_redact_fields, extra_schema_urls)),
+            additional_imports=["typing.ClassVar", "openlineage.client.constants.DEFAULT_PRODUCER"],
+        )
+
+        # keep information about uuid and date-time formats
+        # this is going to be changed back to str type hint in jinja template
+        uuid_type = copy.deepcopy(parser.data_type_manager.type_map[Types.uuid])
+        uuid_type.type = "uuid"
+        parser.data_type_manager.type_map[Types.uuid] = uuid_type
+
+        date_time_type = copy.deepcopy(parser.data_type_manager.type_map[Types.date_time])
+        date_time_type.type = "date-time"
+        parser.data_type_manager.type_map[Types.date_time] = date_time_type
+
+        parser.parse(format_=False)
+
+        # parse rest of spec
+        parser.source = temporary_locations[1:]
+
+        # change paths so that parser sees base objects as local
+        parser.model_resolver.references = {
+            k.replace("OpenLineage.json", base_ids["OpenLineage.json"]): v
+            for k, v in parser.model_resolver.references.items()
+        }
+
+        output = parser.parse(format_=False)
+
+    # go back to client location
+    os.chdir(python_client_location)
+
+    return output
+
+
+def format_and_save_output(output: str, location: pathlib.Path):
+    """Adjust and format generated file."""
+    import subprocess
+
+    # save temporary file to the same directory that output
+    # so that ruff receives same rules
+    with tempfile.NamedTemporaryFile(
+        "w", prefix=location.stem.lower(), suffix=".py", dir=default_output_location.parent, delete=False
+    ) as tmpfile:
+        tmpfile.write(header)
+        is_base_module = "from ." not in output
+        tmpfile.write(output.replace("from .OpenLineage", "from openlineage.client.generated.base"))
+        if is_base_module:
+            tmpfile.write(set_producer_code)
+        tmpfile.flush()
+
+    # run ruff lint
+    with subprocess.Popen(
+        args=["ruff", tmpfile.name, "--fix"], stderr=subprocess.STDOUT, close_fds=True
+    ) as lint_process:
+        if lint_process.returncode:
+            print(f"Ruff lint failed: {lint_process.returncode}")
+
+    # run ruff format
+    with subprocess.Popen(
+        args=["ruff", "format", tmpfile.name],
+        stderr=subprocess.STDOUT,
+        close_fds=True,
+    ) as format_process:
+        if format_process.returncode:
+            print(f"Ruff format failed: {format_process.returncode}")
+
+    # move file to output location
+    if location.name == "open_lineage.py":
+        location = location.with_name("base.py")
+    os.rename(tmpfile.name, location)
+
+    return lint_process.returncode or format_process.returncode
```

Comment:
I like the idea of generating it to temp file and then renaming if everything works fine, but will the return codes be checked anywhere? We could probably raise some error ?

Reply:
I'll add logging as mentioned above.